src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcherIntTest.groovy

```diff
@@ -50,19 +57,46 @@ class HealthServiceInstancesWatcherIntTest extends Specification {
         }
 
         when: "unhealthy service instance is registered"
-        consulCluster.registerUnhealthyServiceInstance("my-service", "dc1", "node1-dc1")
+        consulCluster.registerUnhealthyServiceInstance("my-service", GIVEN_DC, GIVEN_NODE)
 
         and: "healthy service instance is registered"
-        consulCluster.registerHealthyServiceInstance("my-service", "dc1", "node1-dc1", ["tag1", "tag2"])
+        consulCluster.registerHealthyServiceInstance("my-service", GIVEN_DC, GIVEN_NODE, ["tag1", "tag2"])
 
         then: "watcher caught new state with only healthy service instance"
         new PollingConditions(timeout: 10).eventually {
             latestState.head().instances.size() == 1
             def instance = latestState.head().instances.first()
             instance.serviceId != null
-            instance.serviceAddress == "localhost"
-            instance.servicePort == 1234
+            instance.serviceAddress.get() == "localhost"
+            instance.servicePort.get() == 1234
             instance.serviceTags == ["tag1", "tag2"]
         }
     }
+
+    def "should watch only parsable instances"() {
+        given: "watcher on my-service details"
+        Deque<ServiceInstances> latestState = new ArrayDeque<>()
+        healthServiceInstancesWatcher.watch(
+                { latestState.push(it.body as ServiceInstances) },
+                { logger.error("Error while watching", it) })
+
+        expect: "watcher caught first empty state of service details"
+        new PollingConditions(timeout: 10).eventually {
+            !latestState.empty
+            latestState.head().instances.empty
+        }
+
+        when: "unhealthy service instance is registered"
+        consulCluster.registerInstanceLackingPortNumber("my-service", GIVEN_DC, GIVEN_NODE, ["tag1", "tag2"])
+
+        then: "watcher caught new state with only healthy service instance"
+        new PollingConditions(timeout: 10).eventually {
+            latestState.head().instances.size() == 1
+            def instanceWithNoPort = latestState.head().instances.first()
+            instanceWithNoPort.serviceId != null
+            instanceWithNoPort.serviceAddress.get() == "localhost"
+            !instanceWithNoPort.servicePort.isPresent()
```

Comment:
```suggestion
            instanceWithNoPort.servicePort.isEmpty()
```

Reply:
groovy has issue with that, and thats why it has been done this way `No signature of method: java.util.Optional.isEmpty() is applicable for argument types: () values: []
Possible solutions: empty(), identity(groovy.lang.Closure), inspect(), dump()
groovy.lang.MissingMethodException: No signature of method: java.util.Optional.isEmpty() is applicable for argument types: () values: []
Possible solutions: empty(), identity(groovy.lang.Closure), inspect(), dump()
	at pl.allegro.tech.discovery.consul.recipes.watch.health.HealthServiceInstancesWatcherIntTest.should watch only parsable instances(HealthServiceInstancesWatcherIntTest.groovy:93)`